### PR TITLE
adds: Material styling to search and results components

### LINKED ISSUE #839: Search Bar submit button + onClick()
**What would you like to be added**:
#792 should hopefully go in soon. It doesn't have a search button, but dynamically returns results based on user input. I made it this way currently because I'm unable to prevent the onClick function from triggering non stop after the initial user click

Here's some code snippets that may speed up the process to resolve the issue:

I removed the following code from `SearchBar.jsx`:
line 90
```
        <IconButton
          type="submit"
          className={classes.iconButton}
          aria-label="search"
          onClick={event => handleClick(event)}
        >
          <SearchIcon />
        </IconButton>
```

Code for `SearchPage.jsx`:
```
  function onClickHandler() {
    executeSearch();
  }
```
line 103
```
      <SearchBar
        onClickHandler={onClickHandler}
        searchText={searchText}
        onChangeHandler={onChangeHandler}
      />
```

**Why would you like this to be added**:
Up for debate, should we search upon user input or when the user clicks on button? If this is not a concern, we can close this issue.

### PR DESCRIPTION
**Pull request by raygervais**
## Issue This PR Addresses

Fixes #839 

## Type of Change

- [X] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [X] **UI**: Change which improves UI

## Description

- Updates Interface to follow Material Design form spec, see screenshot below.
- Updates logic to submit on button click / enter.

## Checklist

- [X] **Quality**: This PR builds and passes our npm test and works locally
- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

Screenshot from 2020-09-27 19-45-32

Screenshot from 2020-10-01 11-04-48

*Resubmitting PR for that sweet Hacktober Participation.*

### CODE REVIEW

**Comment by raygervais** [comment]
Note, images will randomize on deployment, can't be seen via vercel preview.

File: src/frontend/src/components/AuthorResult/AuthorResult.jsx [hunk 1]
```diff
@@ -1,53 +1,68 @@
...
-import Paper from '@material-ui/core/Paper';
-import Grid from '@material-ui/core/Grid';
 import Avatar from '@material-ui/core/Avatar';
-import List from '@material-ui/core/List';
-import ListItem from '@material-ui/core/ListItem';
-import ListItemIcon from '@material-ui/core/ListItemIcon';
-import ListItemText from '@material-ui/core/ListItemText';
+import Container from '@material-ui/core/Container';
+import Card from '@material-ui/core/Card';
+import CardMedia from '@material-ui/core/CardMedia';
```
Comment (cindyledev) [inline src/frontend/src/components/AuthorResult/AuthorResult.jsx:8]: Small nit and I can slam on the Approve button. You can combine these imports into one line
`import { Avatar, Card, CardActionArea, CardContent, CardMedia, Container, Typography } from '@material-ui/core';`

**Review by cindyledev — CHANGES_REQUESTED** [review]

File: src/frontend/src/components/AuthorResult/AuthorResult.jsx (hunk 1 above)
Comment (raygervais) [reply src/frontend/src/components/AuthorResult/AuthorResult.jsx]: Resolved.

**Comment by raygervais** [comment]
@c3ho, this PR may clash with your PR #1143. Thoughts?

**Review by cindyledev — APPROVED** [review]

**Comment by c3ho** [comment]
@raygervais It will, but I can rebase, won't be too much work on my end. There's no functional changes right? I tested it on Vercel and the search still triggered with `onChange` instead of a `onClick`

**Comment by raygervais** [comment]
Correct, no new functional change aside from minor clean up.

**Review by manekenpix — APPROVED** [review]
Really nice! :+1:

**Comment by c3ho** [comment]
Quick thing I want to check, can you try searching by Post, try searching for Lab1 (there should be some results), then click on search button. It looks like it causes the results to re-render again after loading once?

**Comment by raygervais** [comment]
I still need to work on the avatar initials, will do that in a small follow up @c3ho.

**Comment by c3ho** [comment]
Issue #1088 may need to remove the cardAction this PR implements

**Pull request merged**